### ISSUE TITLE
Data not saving or publishing

### ISSUE THREAD

**Issue by cyberangel67** [thommcgrath/Beacon#313]
I am really at my wits end with this and do not know what I am doing wrong.

I am trying to fix the data in my Mod that has been published and I have spent 2 hours and nothing has saved. I periodically saved it every 10 mins, because of the last bug I raised where if it is open too long it gets stuck in a save loop. Which was closed for some reason.

Anyway, here is what I did.

I replaced all the Blueprint paths with the correct paths, added the new blueprints, and then exported the data. I then tried to publish it and it failed. I then closed Beacon to reopen it back up, reimported the exported data and all the work I have done is gone.

Not only that, but it tells me the published data was last published in May, some 4 weeks ago.

What have I done wrong? Is my data lost again, do I need to actually start from scratch or what is the deal here?

**Comment by thommcgrath** [comment]
I'm looking at the data published for your mod, but it seems to be published. Aside from the fact that most of the item paths are pointing to engram files, what kind of updates need to be made? If you'd like, you can attach the exported beacondata file here so I can take a closer look.

<img width="1312" alt="SCR-20240616-kwyq" src="https://github.com/thommcgrath/Beacon/assets/12851521/dfc1a673-3f77-43c7-ab58-251421f3c8a6">

**Comment by thommcgrath** [comment]
Sit tight, I'm going to try to clean this up for you.

**Comment by thommcgrath** [comment]
Ok, here's what I've done. I had taken ownership of the mod for a bit and used a tool to extract data from the mod files on CurseForge. You've got a good data set in there now that I'd say is about 95% of the way to perfect. I'm not very familiar with the mod, so you may need to remove some items that don't really belong or maybe adjust some names. But the paths, unlock strings, point requirements, and recipes are all there.

I'm not going to pretend this wasn't a challenge, because (like you) I definitely found some problems with the process. The chief among them is that importing from a local mod into an official mod generates new UUIDs for everything, but is failing to update the recipe UUIDs accordingly. So items that reference other items in the mod end up breaking, and that's the issue you were running into. Things like the CS Tek Stove requiring the CS Industrial Grill were trying to reference the wrong object, despite looking correct in the editor. To make matters worse, Beacon is unable to find mod items in unsaved mods since they aren't actively in the database yet. This is a tough challenge to solve, but I've got some of a solution in a branch at the moment. I'm still working on getting it ready for production.

So anyway, I've found issues I need to work on in Beacon, and the data published for your mod is up to date. I've transferred ownership of the mod back over to you.

**Comment by thommcgrath** [comment]
No, I wasn't expecting any more issues. What exactly did you change? I'm not seeing any error messages in my database logs either, so I can't really tell what is failing.

**Comment by thommcgrath** [comment]
If you have export selected, export will export everything. Otherwise it'll export what you have selected. Publish is more like a save operation, and it will always send all changes to the server.
